bfcfg

```diff
@@ -1303,40 +1873,489 @@ sb_cfg()
+          bmc_cfg_get_value ${out_file} "BMC_LOCK_ACCOUNT_MAX_RETRY" "$(( $off + 6 ))" 2 ${in_file}
+          bmc_cfg_get_value ${out_file} "BMC_SOL_ENABLE" "$(( $off + 8 ))" 1 ${in_file}
+          bmc_cfg_get_value ${out_file} "BMC_SOL_CHARACTER_SEND_THRESHOLD" "$(( $off + 9 ))" 2 ${in_file}
+          bmc_cfg_get_value ${out_file} "BMC_SOL_CHARACTER_ACCUMULATE_LEVEL" "$(( $off + 11 ))" 2 ${in_file}
+          bmc_cfg_get_value ${out_file} "BMC_WEBUI_ENABLE" "$(( $off + 16 ))" 1 ${in_file}
+          off=$(( $off + $BMC_CFG_SUBTYPE_MISC_LEN))
+        
+        elif [ $subtype -eq "$BMC_CFG_SUBTYPE_USER_DEFAULT" ]; then
+          idx=$(to_integer ${in_file} $(( $off + 3 )) 1)
+          bmc_cfg_get_value ${out_file} "BMC_USER_DEFAULT_PRIVILEGE_${idx}" "$(( $off + 4 ))" 56 ${in_file}
+          bmc_cfg_get_value ${out_file} "BMC_USER_DEFAULT_USERNAME_${idx}" "$(( $off + 60 ))" 64 ${in_file}
+          bmc_cfg_get_value ${out_file} "BMC_USER_DEFAULT_PASSWORD_${idx}" "$(( $off + 124 ))" 128 ${in_file}
+          off=$(( $off +  $BMC_CFG_SUBTYPE_USER_DEFAULT_LEN))
+
+        elif [ $subtype -eq "$BMC_CFG_SUBTYPE_NTP" ]; then
+          idx=$(to_integer ${in_file} $(( $off + 7 )) 1)
+          bmc_cfg_get_value ${out_file} "BMC_NTP_${idx}" "$(( $off + 8 ))" 256 ${in_file}
+          off=$(( $off + $BMC_CFG_SUBTYPE_NTP_LEN))
+        
+        fi
+
+        [[ $off -ge $limit ]] && break
+      done
+
+    else
+        # Type nt supported, thus skip!
+        off=$(( $off + $length ))
+
+    fi
+
+    [[ $off -ge $in_file_len ]] && break
+  done
+
+  return
+}
+
+#
+# Fixed parameters for BFB and capsule files.
+# MUST be consistent with the definition of BFB header
+# format and UEFI capsule header format. 
+#
+# The BFB header magic number "BF^B^S" (0x13026642) 
+BFB_MAGIC="42660213"
+# Unsigned capsule header length.
+CAP_HEADER_LEN=168
+CAP_GUID="edd5cb6d2de8444cbda17194199ad92a"
+
+# This function doesn't support signed capsule files.
+parse_cap()
+{
+  incap=$1
+  outbin=$2
+
+  # Check UEFI capsule GUID
+  #local capguid=$(tail -c+1 ${bin_file} | head -c16 | xxd -p)
+  local capguid=$(dd if=${incap} skip=0 count=16 bs=1 2> /dev/null | xxd -p)
+  [ "$capguid" != "$CAP_GUID" ] && return
+  
+  log_msg "bin2cfg: CAP file detected."
+
+	echo "Parsing EFI Capsule $incap..."
+
+	# Convert capsule length in hex to integer.
+	capfile_len=$(stat -c%s "$incap")
+	# Quick check on the capsule length.
+	if [ $capfile_len -le $CAP_HEADER_LEN ]; then
+		echo "** error: invalid capsule file length!"
+		return
+	fi
+
+	# Calculate capsule header size. The header is skipped.
+  # @FIXME: Determine header length from capsule.
+	capfile_skip=$CAP_HEADER_LEN
+	# Extract capsule payload. 
+	dd if=${incap} of=${outbin} bs=1 count=$(( $capfile_len - $CAP_HEADER_LEN )) skip=$capfile_skip 2> /dev/null
+
+  return
+}
+
+parse_bfb()
+{
+  local inbfb=$1
+  local outbin=$2
+  local tmp_cap_file='/tmp/.bf.cfg.cap.tmp'
+
+  local bfb_hdr_len=24
+	#
```

Comment:
check indentation in this function - a few areas appear to be offset from surrounding code

Reply:
Very weird, I don't know how it happened. maybe a copy paste or something. I'll fix it.